packages/flutter/lib/src/widgets/animated_scroll_view.dart

```diff
@@ -211,6 +211,341 @@ class AnimatedListState extends _AnimatedScrollViewState<AnimatedList> {
+///         );
+///       },
+///       separatorBuilder: (BuildContext context, int index, Animation<double> animation) {
+///         return const Divider();
+///       }
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [SliverAnimatedList], a sliver that animates items when they are inserted
+///    or removed from a list.
+///  * [SliverAnimatedGrid], a sliver which animates items when they are
+///    inserted or removed from a grid.
+///  * [AnimatedGrid], a non-sliver scrolling container that animates items when
+///    they are inserted or removed in a grid.
+class AnimatedListSeparated extends _AnimatedScrollView {
+  /// Creates a scrolling container that animates items with separators when
+  /// they are inserted or removed.
+  AnimatedListSeparated({
+    super.key,
+    required AnimatedItemBuilder itemBuilder,
+    required AnimatedItemBuilder separatorBuilder,
+    int initialItemCount = 0,
+    super.scrollDirection = Axis.vertical,
+    super.reverse = false,
+    super.controller,
+    super.primary,
+    super.physics,
+    super.shrinkWrap = false,
+    super.padding,
+    super.clipBehavior = Clip.hardEdge,
+  }) : assert(initialItemCount >= 0),
+       super(
+         initialItemCount: _computeChildCountWithSeparators(initialItemCount),
+         itemBuilder: (BuildContext context, int index, Animation<double> animation) {
+           final int itemIndex = index ~/ 2;
+           if (index.isEven) {
+             return itemBuilder(context, itemIndex, animation);
+           }
+           return separatorBuilder(context, itemIndex, animation);
+         },
+       );
+
+  /// The state from the closest instance of this class that encloses the given
+  /// context.
+  ///
+  /// This method is typically used by [AnimatedListSeparated] item widgets that insert
+  /// or remove items in response to user input.
+  ///
+  /// If no [AnimatedListSeparated] surrounds the context given, then this function will
+  /// assert in debug mode and throw an exception in release mode.
+  ///
+  /// This method can be expensive (it walks the element tree).
+  ///
+  /// This method does not create a dependency, and so will not cause rebuilding
+  /// when the state changes.
+  ///
+  /// See also:
+  ///
+  ///  * [maybeOf], a similar function that will return null if no
+  ///    [AnimatedListSeparated] ancestor is found.
+  static AnimatedListSeparatedState of(BuildContext context) {
+    final AnimatedListSeparatedState? result = AnimatedListSeparated.maybeOf(context);
+    assert(() {
+      if (result == null) {
+        throw FlutterError.fromParts(<DiagnosticsNode>[
+          ErrorSummary('AnimatedListSeparated.of() called with a context that does not contain an AnimatedListSeparated.'),
+          ErrorDescription(
+            'No AnimatedListSeparated ancestor could be found starting'
+            ' from the context that was passed to AnimatedListSeparated.of().',
```

Comment:
nit: we prefer having the space on the end of the line before the cut

(if this is copy/pasted from some other code, bonus points for fixing that code too)